pkg/http/server.go

```diff
@@ -22,5 +22,9 @@ func ServeContext(ctx context.Context, addr string, hdl http.Handler) error {
 		}
 	}()
 
+	if tlsEnabled {
+		return srv.ListenAndServeTLS(tlsConfig["certificatePath"], tlsConfig["keyPath"])
+	}
+
 	return srv.ListenAndServe()
```

Comment:
Shouldn't we allow both? I mean, will we enforce that all the traffic uses HTTPS or HTTP?  I'm not directly against this because we have in the roadmap the plan of scoped interceptors, but I prefer to expose both and route the request based on the incoming protocol

Reply:
We could but I can't see a use case for supporting both simultaneously - unless maybe during a migration scenario? Do you have any examples where you could see serving both and routing based on the protocol to be beneficial?

Reply:
The problem is that currently, the interceptor is shared. If you have a cluster with multiple workloads, you could be interested on both scenarios, but it's true that probably you won't be interested on having both during a long time, just during migrations 🤔 

Maybe we can continue with the PR as it is (just supporting one of them) and work on dedicated interceptors before next release. This can allow both scenarios easily
WDYT @tomkerkhove ?

Reply:
I agree exposing both would be good to offer

Reply:
I guess that we can go with this as it is and add it as nice to have before GA.